### ISSUE TITLE
Feature Request: Support for ECS IPC Mode/PID Mode

### ISSUE THREAD

**Issue by gazoakley** [terraform-providers/terraform-provider-aws#6511]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon Elastic Container Service now allows specifying flags for `pidMode` and `ipcMode`:

> The `pidMode` parameter allows you to configure your containers to share their process ID (PID) namespace with other containers in the task, or with the host. Sharing the PID namespace enables for example monitoring applications deployed as containers to access information about other applications running in the same task or host.
>
> The `ipcMode` parameter allows you to configure your containers to share their inter-process communication (IPC) namespace with the other containers in the task, or with the host. The IPC namespace allows containers to communicate directly through shared-memory with other containers running in the same task or host.

### New or Affected Resource(s)

* aws_ecs_task_definition

### Potential Terraform Configuration

```hcl
resource "aws_ecs_task_definition" "service" {
  family                = "service"
  container_definitions = "${file("task-definitions/service.json")}"

  ipc_mode = "task"
  pid_mode = "host"
}
```

### References

* https://aws.amazon.com/about-aws/whats-new/2018/10/amazon-ecs-now-allows-two-additional-docker-flags-/
* https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#other_task_definition_params

**Comment by gazoakley** [comment]
Requires #6513

**Comment by bflad** [comment]
Support for `ipc_mode` and `pid_mode` arguments for the `aws_ecs_task_definition` resource has been merged and will release with version 1.46.0 of the AWS provider, likely later today or tomorrow.

**Comment by gazoakley** [comment]
Oh - I had a branch for this already sorted locally 🙁

**Comment by bflad** [comment]
This has been released in [version 1.46.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1460-november-20-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
